src/components/Text/Text.styles.ts

```diff
@@ -1,38 +1,55 @@
 /*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
-import { TextProps, TextSize } from './Text';
-import {
-  ellipsisStyle,
-  fontWeights,
-  fontSizes,
-  lineHeights,
-  textColors,
-  verticalAligns,
-} from '../../util/styles/fonts';
-import { IRawStyle } from '@uifabric/styling';
+import { TextProps } from './Text';
+import { TextSize } from './types';
+import { memoizeFunction } from '@uifabric/utilities';
+import { mergeStyleSets } from '@uifabric/styling';
+import { ellipsisStyle, fontWeights, textColors, verticalAligns } from '../../util/styles/fonts';
+import { getTheme } from '../../util/colors';
 
-const getHeight = (size?: TextSize) => {
-  return size ? lineHeights[size] : undefined;
+const iconSizeForSize = {
+  [TextSize.XXLARGE]: '2.4rem',
+  [TextSize.XLARGE]: '2.2rem',
+  [TextSize.LARGE]: '1.5rem',
+  [TextSize.MEDIUM]: '1.4rem',
+  [TextSize.MEDIUM_SUB]: '1.4rem',
+  [TextSize.SMALL]: '1.2rem',
+  [TextSize.XSMALL]: '1.0rem',
 };
 
-export const getStyles = (props: TextProps): IRawStyle => {
-  const { size, maxWidth, bold, uppercase, color } = props;
+const getMemoizedClassNames = memoizeFunction((styleProps: TextProps, theme) => {
+  const { size, maxWidth, bold, uppercase, color } = styleProps;
+  const font = size ? theme.fonts[size === TextSize.MEDIUM_SUB ? 'smallPlus' : size] : undefined;
```

Comment:
Todo in a future PR: rename MEDIUM_SUB to SMALL_PLUS to align with Fabric.
https://github.com/Microsoft/YamUI/issues/424